cmd/clusterctl/pkg/client/cluster/move.go

```diff
@@ -202,11 +232,397 @@ func (o *objectGraph) Discovery(namespace string, types []metav1.TypeMeta) error
+				errOnce.Do(func() { gerr = err })
+			}
+		}(dependent)
+	}
+
+	for dependent := range nodeToCreate.softDependents {
+		wg.Add(1)
+		go func(dependent *node) {
+			defer wg.Done()
+			if err := o.createTreeIntoTarget(nodeToCreate, dependent); err != nil {
+				errOnce.Do(func() { gerr = err })
+			}
+		}(dependent)
+	}
+
+	wg.Wait()
+	if gerr != nil {
+		return gerr
+	}
+
+	// If the nodeToCreate is a machine, wait for the machine to be reconciled and linked with the corresponding node.
+	if nodeToCreate.identity.APIVersion == clusterv1.GroupVersion.String() && nodeToCreate.identity.Kind == "Machine" {
+		o.log.V(1).Info("Waiting for reconciliation with the workload cluster node", "Machine", nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+		machineKey := client.ObjectKey{Namespace: nodeToCreate.identity.Namespace, Name: nodeToCreate.identity.Name}
+		if err := o.machineWaiter(o.toProxy, machineKey); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createTargetObject creates the Kubernetes object corresponding to the node in the target Management cluster, taking care of restoring the OwnerReference with the owner node, if any.
+func (o *objectGraph) createTargetObject(ownerNode, nodeToCreate *node) error {
+	o.log.V(1).Info("Creating", nodeToCreate.identity.Kind, nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+
+	cFrom, err := o.fromProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Get the source object
+	sourceObj := &unstructured.Unstructured{}
+	sourceObj.SetAPIVersion(nodeToCreate.identity.APIVersion)
+	sourceObj.SetKind(nodeToCreate.identity.Kind)
+	targetObjKey := client.ObjectKey{
+		Namespace: nodeToCreate.identity.Namespace,
+		Name:      nodeToCreate.identity.Name,
+	}
+
+	if err := cFrom.Get(ctx, targetObjKey, sourceObj); err != nil {
+		return errors.Wrapf(err, "error reading %q %s/%s",
+			sourceObj.GroupVersionKind(), sourceObj.GetNamespace(), sourceObj.GetName())
+	}
+
+	targetObj := sourceObj.DeepCopy()
+
+	// New objects cannot have a specified resource version. Clear it out.
+	targetObj.SetResourceVersion("")
+
+	// Removes current OwnerReferences
+	targetObj.SetOwnerReferences(nil)
+
+	// Creates a new OwnerReferences towards the OwnerNode, if any.
+	if ownerNode != nil {
+		ownerRef := metav1.OwnerReference{
+			APIVersion: ownerNode.identity.APIVersion,
+			Kind:       ownerNode.identity.Kind,
+			Name:       ownerNode.identity.Name,
+			UID:        ownerNode.newUID, // Use the owner's newUID read from the target management cluster (instead of the UID read during discovery).
+		}
+
+		// Restores the attributes of the OwnerReference.
+		if attributes, ok := ownerNode.dependents[nodeToCreate]; ok {
+			ownerRef.Controller = attributes.Controller
+			ownerRef.BlockOwnerDeletion = attributes.BlockOwnerDeletion
+		}
+
+		targetObj.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
```

Comment:
The goal of this iteration is to support to support descendant discovery via owner refs, which is blocking for using the move operation with CAPV (rif https://github.com/kubernetes-sigs/cluster-api/issues/1525#issuecomment-571164648).

The limit of a single owner still exists (like in the former move), but we are in better shape to get this addressed as well, even if I'm not sure this can be done for the RC.
AFAIK a *Templates might have many owners if shared between many clusters, but I consider this a nice to have/not a regression vs the current state.